controllers/subscriptions.go

```diff
@@ -102,6 +114,23 @@ func CheckExistingSubscriptions(cli client.Client, desiredSubscription *operator
 	return desiredSubscription, nil
 }
 
+func isProviderServer(cli client.Client) (bool, error) {
```

Comment:
not sure `isProviderServer` sounds correct, to be abstract can use `isProviderMode` or to be exact can use `isAllowRemoteStorageConsumersSet`

Reply:
I will change it to `isProviderMode` as we always refer it as provider.